src/main/java/org/logstash/filters/GeoIPFilter.java

```diff
@@ -0,0 +1,342 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.logstash.filters;
+
+import com.maxmind.db.CHMCache;
+import com.maxmind.db.InvalidDatabaseException;
+import com.maxmind.geoip2.exception.AddressNotFoundException;
+import com.maxmind.geoip2.exception.GeoIp2Exception;
+import com.maxmind.geoip2.model.CityResponse;
+import com.maxmind.geoip2.model.CountryResponse;
+import com.maxmind.geoip2.model.IspResponse;
+import com.maxmind.geoip2.record.*;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.logstash.Event;
+
+import com.maxmind.geoip2.DatabaseReader;
+import org.logstash.ext.JrubyEventExtLibrary.RubyEvent;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.*;
+
+public class GeoIPFilter {
+  private static Logger logger = LogManager.getLogger();
+  private static final String CITY_DB_TYPE = "GeoLite2-City";
+  private static final String COUNTRY_DB_TYPE = "GeoLite2-Country";
```

Comment:
The commercial databases have the type `GeoIP2-City` and `GeoIP2-Country`. It'd likely be worth supporting those too, especially given that they are mentioned in the [recent blog post](https://www.elastic.co/blog/geoip-in-the-elastic-stack).